### PR TITLE
Arabic: Add new local datasets, starting by ar-dz

### PR DESCRIPTION
**Pull request by linuxscout**
# I have made things!

## Checklist

- [x] I have read [contributing guidelines](https://github.com/lk-geimfari/mimesis/blob/master/CONTRIBUTING.rst)
- [x] I'm sure that I did not unrelated changes in this pull request
- [x] I have created at least one test case for the changes I have made

- [x] I have added myself to the [CONTRIBUTORS.rst](https://github.com/lk-geimfari/mimesis/blob/master/CONTRIBUTORS.rst)

### CODE REVIEW

File: mimesis/datasets/ar/address.json [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+ 
+  "عين الأربعاء",
+  "عين الإبل",
+   "عين الباردة",
+  "عين البرد",
+  "عين البنيان",
+   "عين البيضاء",
+    "عين الترك",
+    "عين التركي",
```
Comment (lk-geimfari) [inline mimesis/datasets/ar/address.json:14]: I know Arabic is a right-to-left (RTL) language, but the indentation seems incorrect here. Could you please format all these files using JSON Online Editor or a similar tool?

File: mimesis/datasets/ar/datetime.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
+{
+"day": {
+    "abbr": [
+ "س",
```
Comment (lk-geimfari) [inline mimesis/datasets/ar/datetime.json:4]: Same as above here.

**Comment by lk-geimfari** [comment]
Please fix the tests and issues mentioned above, and then this PR will be ready to merge.

**Comment by linuxscout** [comment]
> Also, it would be great to add Algeria to this list: https://github.com/lk-geimfari/mimesis/blob/master/docs/locales.rst#supported-locales

How to insert flag?

**Comment by linuxscout** [comment]
> Also, it would be great to add Algeria to this list: https://github.com/lk-geimfari/mimesis/blob/master/docs/locales.rst#supported-locales

ok, done.

File: mimesis/datasets/ar/datetime.json (hunk 2 above)
Comment (linuxscout) [reply mimesis/datasets/ar/datetime.json]: ok, done

File: mimesis/datasets/ar/address.json (hunk 1 above)
Comment (linuxscout) [reply mimesis/datasets/ar/address.json:11]: ok, done

File: mimesis/enums.py [hunk 3]
```diff
@@ -65,7 +65,23 @@ class Locale(Enum):
 
     An argument for all local-depend providers.
     """
-
+    AR_AE = "ar-ae"
+    # AR_BH = "ar-bh"
```
Comment (lk-geimfari) [inline mimesis/enums.py:69]: Please, remove these commented locales.

**Review by lk-geimfari — CHANGES_REQUESTED** [review]

File: mimesis/enums.py (hunk 3 above)
Comment (linuxscout) [reply mimesis/enums.py]: ok, removed.

**Comment by lk-geimfari** [comment]
Well, we’re almost done. Can you please address the failed tests?

To fix the formatting, run the following command:

```
make format
```

**Comment by lk-geimfari** [comment]
> Is there anything to do?

That's it. Thank you. I'll merge this PR on this weekend and release new version soon.

**Comment by lk-geimfari** [comment]
@linuxscout https://github.com/lk-geimfari/mimesis/releases/tag/v18.0.0

Thank you for your contribution.

**Comment by lk-geimfari** [comment]
@linuxscout Can you please look at this test fail:

https://github.com/lk-geimfari/mimesis/actions/runs/10856979200/job/30132622505

Unfortunately it's hard for me to understand what's wrong as I don't know the language.

**Pull request merged**